src/Sylius/Bundle/MailerBundle/Renderer/Adapter/EmailTwigAdapter.php

```diff
@@ -62,17 +62,39 @@ public function render(EmailInterface $email, array $data = [])
     private function getRenderedEmail(EmailInterface $email, array $data)
     {
         if (null !== $email->getTemplate()) {
-            $data = $this->twig->mergeGlobals($data);
+            return $this->provideEmailWithTemplate($email, $data);
+        }
 
-            /** @var \Twig_Template $template */
-            $template = $this->twig->loadTemplate($email->getTemplate());
+        return $this->provideEmailWithoutTemplate($email, $data);
+    }
 
-            $subject = $template->renderBlock('subject', $data);
-            $body = $template->renderBlock('body', $data);
+    /**
+     * @param EmailInterface $email
+     * @param array $data
+     *
+     * @return RenderedEmail
+     */
+    private function provideEmailWithTemplate(EmailInterface $email, array $data)
+    {
+        $data = $this->twig->mergeGlobals($data);
 
-            return new RenderedEmail($subject, $body);
-        }
+        /** @var \Twig_Template $template */
+        $template = $this->twig->loadTemplate($email->getTemplate());
+
+        $subject = $template->renderBlock('subject', $data);
+        $body = $template->renderBlock('body', $data);
 
+        return new RenderedEmail($subject, $body);
+    }
+
+    /**
+     * @param EmailInterface $email
+     * @param array $data
+     *
+     * @return RenderedEmail
+     */
+    private function provideEmailWithoutTemplate(EmailInterface $email, array $data)
+    {
         $twig = new \Twig_Environment(new \Twig_Loader_Array([]));
```

Comment:
Probably because there are lot of bad things happening in *MailerBundle* 😄  Refactoring incoming 🐃